instance/models/openedx_instance.py

```diff
@@ -198,6 +201,28 @@ def database_name(self):
         escaped = ''.join(char for char in name if char in allowed)
         return truncate_name(escaped, length=50)
 
+    @property
+    def shutdown(self):
+        """
+        Return True if this instance has been shut down, else False.
+
+        An instance has been shut down if all of its app servers have been terminated,
+        and monitoring has been turned off.
+
+        If an instance has no app servers, we assume that it has *not* been shut down.
+        This ensures that the GUI lists newly created instances without app servers.
+        """
+        if self.appserver_set.count() == 0:
+            return False
+        all_appservers_terminated = all(
```

Comment:
@haikuginger I think that mixing app server states with properties that provide state(-related) information has the potential to be confusing, so I didn't add an `is_shutdown` property to the app server. But I did move the logic from this block to a separate function to be able to simplify the generator expression (which now looks almost identical to what you were suggesting). See 6e9255b7457431601977d65faea992788e8ec7d0.